pkg/client/cache/reflector.go

```diff
@@ -337,16 +304,23 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 			return nil
 		}
-		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+
+		if err := r.watchHandler(w, &resourceVersion, stopCh); err != nil {
+			if err != errorStopRequested {
 				glog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedType, err)
 			}
 			return nil
 		}
-		if r.canForceResyncNow() {
-			glog.V(4).Infof("%s: next resync planned for %#v, forcing now", r.name, r.nextResync)
-			return nil
-		}
+		/*
```

Comment:
This is something I'm not 100% sure we want to remove. Basically, now all watch connections has timeout in order of small minutes. If we get rid of that, and we have some network issue, we may end up with watch that is broken because of network issue and will never be closed and renewed.
So I would prefer leaving this for now (and potentially add a TODO to consider removing it in the future).